src/parsers/rdp.rs

```diff
@@ -237,13 +237,21 @@ macro_rules! impl_rdp {
                 self.fail_pos = 0;
             }
 
+            fn queue_debug(&self) -> Vec<(Token<Rule>, String)> {
+                self.queue
+                    .clone()
+                    .into_iter()
+                    .map(|t| (t, self.input().slice(t.start, t.end).to_string()))
```

Comment:
Even though `to_string` has been optimized in the latest versions, I'd still use `to_owned` since it makes more sense in terms of terminology.